src/steam_network/protocol/protobuf_client.py

```diff
@@ -624,38 +615,32 @@ async def _process_client_persona_state(self, body: bytes):
         if self.user_info_handler is None:
             return
 
-        message = CMsgClientPersonaState()
-        message.ParseFromString(body)
+        message = CMsgClientPersonaState().parse(body)
 
         for user in message.friends:
             user_id = user.friendid
-            if user_id == self.confirmed_steam_id and int(user.game_played_app_id) != 0:
-                await self.get_apps_info([int(user.game_played_app_id)])
-            user_info = ProtoUserInfo()
-            if user.HasField("player_name"):
-                user_info.name = user.player_name
-            if user.HasField("avatar_hash"):
-                user_info.avatar_hash = user.avatar_hash
-            if user.HasField("persona_state"):
-                user_info.state = EPersonaState(user.persona_state)
-            if user.HasField("gameid"):
-                user_info.game_id = user.gameid
-                rich_presence: Dict[str, str] = {}
-                for element in user.rich_presence:
-                    if type(element.value) == bytes:
-                        logger.warning(f"Unsupported presence type: {type(element.value)} {element.value}")
-                        rich_presence = {}
-                        break
-                    rich_presence[element.key] = element.value
-                    if element.key == 'status' and element.value:
-                        if "#" in element.value:
-                            await self.translations_handler(user.gameid)
-                    if element.key == 'steam_display' and element.value:
-                        if "#" in element.value:
-                            await self.translations_handler(user.gameid)
-                user_info.rich_presence = rich_presence
-            if user.HasField("game_name"):
-                user_info.game_name = user.game_name
+            if user_id == self.confirmed_steam_id and user.game_played_app_id != 0:
+                await self.get_apps_info([user.game_played_app_id])
+
+            user_info = ProtoUserInfo(
+                name=user.player_name,
+                avatar_hash=user.avatar_hash,
+                state=EPersonaState(user.persona_state),
+                game_id=user.gameid
+            )
+
+            rich_presence: Dict[str, str] = {}
+            for element in user.rich_presence:
+                rich_presence[element.key] = element.value
+                if element.key == 'status' and element.value:
+                    if "#" in element.value:
+                        await self.translations_handler(user.gameid)
+                if element.key == 'steam_display' and element.value:
+                    if "#" in element.value:
+                        await self.translations_handler(user.gameid)
+
+            user_info.rich_presence = rich_presence
+            user_info.game_name = user.game_name
```

Comment:
This could be an issue, but i'm not sure why they checked these fields but not other ones. Newer betterproto releases provide more things, but this is not one of them. 

I've thought up a clever, yet dirty workaround for this - python does not enforce type-checking at runtime. we can set all these fields that the old code checked to `None` before calling parse. Since parse only updates fields (it doesn't clear the fields of any existing data), anything that remains "None" is unest.

Reply:
but it doesn't check hasname for presence, if game id passes. wtf?

Reply:
Mentioned this at https://github.com/ABaumher/galaxy-integration-steam/discussions/17#discussioncomment-6141214
If this won't have any effect writing to a cache later, assuming it does that, this doesn't require changes. let me know.

Reply:
There's two things that played into this change: first, native types (int, float, str, etc) are non-nullable in Python so there's always a value and we can safely copy it (at the very last it's the default value for the type so no harm done). Second, `HasField` is specific to google-protobuf, betterproto just doesn't support this type of check ([well, technically they do but only under very specific circumstances](https://github.com/danielgtaylor/python-betterproto#determining-if-a-message-was-sent)): google-protobuf creates message definitions dynamically at runtime and fields are objects within the message class (and hence can be null), in betterproto, however, fields mostly have pyton-native types which are non-nullable and hence the HasField (which is a null check more or less) cannot be implemented for those types. See the above link for more info on that.

Reply:
I'm assuming betterproto uses `default` for the various native types. The issue wasn't there so much as the `ProtoUserInfo` in SteamTypes that uses Optional. In the old code, if the HasName failed, these values were left as `None`, the new code will use `default`. I don't know if this change in value affects other code that interacts with ProtoUserInfo (i.e. if a check says `if protouserinfo_instance.whatever_field is not None`)

Reply:
I've found one example where we do a check on that proto info: https://github.com/ABaumher/galaxy-integration-steam/blob/master/src/steam_network/presence.py#L51-#L96

I think it's ok with the updated code you wrote, but i also have no idea what it's doing.

Reply:
So, going by where that's called, GOG keeps what they call UserPresence on your friends, which displays "in game/away/etc". ProtoUserInfo is what we get from steam. we store that data in memory but afaik don't cache it to the database (it'd make no sense to be cached there since it's real-time info). Whenever GOG asks us for that info, we convert the proto info (which is regularly updated by steam) to presence and pass it to GOG. I think that's the only instance it's used, and as long as that function i linked above handles whatever betterproto defaults to now, we're fine. Sorry for being such a pain in the neck merging this, but it's the little things like this that come back to bite you in the ass.

Reply:
All fields default to whatever the python-default for a given type is (numbers -> 0, bool -> False, str -> "", objects -> None, etc). betterproto does that by [inspecting type hints](https://github.com/danielgtaylor/python-betterproto/blob/master/src/betterproto/__init__.py#L886) within the protoc-generated dataclasses (that's btw the reason why I needed to remove the TYPE_CHECKING from all the files; betterproto would otherwise suddenly break during message parsing because of missing types). Unfortunately, betterproto's docs explicitly state that `You cannot check if a scalar was sent on the wire.` so we must work around that if we need to pass on the information whether a field was actually present in a message (instead of just being having a default value).

Reply:
the workaround would be to set the values in the class to `None` after betterproto creates it but before we call parse. Parse ignores any values not in  the message, so those values would still be `None` after calling parse. It would undoubtedly cause headaches with lists though, so it's not a good solution. I think all i needed to do was update the Presence.py function to check for an empty dictionary and we are fine. I did that on your branch and was about to mark this resolved.